examples/analogin/main.c

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2020-2021 Arm Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define MBED_CONF_MBED_TRACE_ENABLE 1
+#define MBED_TRACE_MAX_LEVEL TRACE_LEVEL_DEBUG
+#define TRACE_GROUP "main"
+
+#include "analogin_api.h"
+#include "serial_api.h"
+#include "mbed_trace.h"
+
+#include <stdio.h>
+
+extern serial_t stdio_uart;
+
+void serial_print(const char *string)
+{
+    while (*string != 0) {
+        serial_putc(&stdio_uart, *string++);
+    }
+    serial_putc(&stdio_uart, '\n');
+}
+
+analogin_t analog_input;
+
+float stored_value;
+float val;
+char out_string[50];
+
+int main()
+{
+    serial_init(&stdio_uart, CONSOLE_TX, CONSOLE_RX);
+    serial_baud(&stdio_uart, MBED_CONF_PLATFORM_STDIO_BAUD_RATE);
+
+    mbed_trace_init();
+    mbed_trace_print_function_set(serial_print);
+
+    analogin_init(&analog_input, ARDUINO_UNO_A0);
+
+    tr_info("Starting analog in example.");
+
+    while (1){
+      stored_value = analogin_read(&analog_input);
+      val = stored_value * 3.3;
+      sprintf(out_string, "The voltage on the A0 pin is %u.%02uV", ((uint16_t)(val * 100.)) / 100, (((uint16_t)(val * 100.)) % 100));
```

Comment:
`sprintf` may overflow the buffer and lead to dramatic events.
Also, floating point operations are fairly expensive and should be factorized.

```suggestion
      uint16_t uval = (uint16_t)(val * 100.);
      int len = snprintf(out_string, 50, "The voltage on the A0 pin is %u.%02uV", uval / 100, uval % 100);
```

Note: /100 & %100 can be factorized with `div` from the stdlib.